services/secondary-storage/controller/src/main/java/org/apache/cloudstack/secondarystorage/SecondaryStorageManagerImpl.java

```diff
@@ -1251,13 +1194,12 @@ public boolean finalizeCommandsOnStart(Commands cmds, VirtualMachineProfile prof
 
         if (controlNic == null) {
             if (managementNic == null) {
-                s_logger.error("Management network doesn't exist for the secondaryStorageVm " + profile.getVirtualMachine());
+                s_logger.warn(String.format("Management network does not exist for the secondary storage %s.", profile. toString()));
                 return false;
             }
             controlNic = managementNic;
         }
 
-        // verify ssh access on management nic for system vm running on HyperV
```

Comment:
how about making this a private method name
```suggestion
        verifySshAccessOnManagementNicForSystemvmRunningOnHyperV(profile, controlNic, managementNic, cmds);
        return true;
    }

    private void verifySshAccessOnManagementNicForSystemvmRunningOnHyperV(VirtualMachineProfile profile,
    NicProfile controlNic,
    NicProfile managementNic,
    Commands cmds) {
        if(profile.getHypervisorType() == HypervisorType.Hyperv) {
            controlNic = managementNic;
        }
        CheckSshCommand check = new CheckSshCommand(profile.getInstanceName(),controlNic.getIPv4Address(), 3922);
        cmds.addCommand("checkSsh", check);
    }
```
context: see comment removal below.

Reply:
@DaanHoogland done. As the comment refers only to `if...` I thought it would be better to only extract it. Also, I extracted as protected and added unit tests to it.

Reply:
tests are always good.
As for the `if...` comment, maybe the lengthy name was not my best effort ever and `verifySshAccessOnManagementNicForSystemvm()` would have been better.

Reply:
@DaanHoogland done.